srtp/srtp.c

```diff
@@ -1723,10 +1728,41 @@ static srtp_err_status_t srtp_protect_aead(srtp_ctx_t *ctx,
      * extension, if present; otherwise, it starts after the last csrc,
      * if any are present
      */
-    enc_start = (uint32_t *)hdr + uint32s_in_rtp_header + hdr->cc;
+
+    /* Cryptex can only encrypt CSRCS if header extension is present*/
+    if (stream->use_cryptex && hdr->cc && !hdr->x) {
+        return srtp_err_status_parse_err;
+    }
     if (hdr->x == 1) {
-        xtn_hdr = (srtp_hdr_xtnd_t *)enc_start;
-        enc_start += (ntohs(xtn_hdr->length) + 1);
+        xtn_hdr = (srtp_hdr_xtnd_t *)(uint32_t *)hdr + uint32s_in_rtp_header +
+                  hdr->cc;
+        xtn_hdr_length = ntohs(xtn_hdr->length);
+        xtn_profile_specific = ntohs(xtn_hdr->profile_specific);
+    }
+    /* If no header extension is present cryptex has no effect */
+    if (stream->use_cryptex && hdr->x) {
+        /* Change profiles by cryptex values */
+        if (xtn_profile_specific == 0xbede) {
+            xtn_hdr_profile_and_value = htonl(0xc0de << 16 | xtn_hdr_length);
+        } else if (xtn_profile_specific == 0x1000) {
```

Comment:
This is incorrect.  The low-order bits don't have to be zero.

cf. https://github.com/cisco/libsrtp/blob/main/srtp/srtp.c#L1638